src/main/java/net/dean/jraw/auth/RefreshTokenHandler.java

```diff
@@ -51,11 +51,18 @@ public boolean isAcquiredTimeStored(String username) {
 
     @Override
     public String readToken(String username) throws NoSuchTokenException {
-        if (reddit.getOAuthData() == null)
+        if (reddit.getOAuthData() == null) {
             return store.readToken(getKeyFor(username));
-        if (reddit.getOAuthData().getRefreshToken() != null)
+
+        } else if (reddit.getOAuthData().getRefreshToken() != null) {
             return reddit.getOAuthData().getRefreshToken();
-        throw new NoSuchTokenException("Refresh token does not exist in the TokenStore nor the RedditClient");
+
+        } else if (reddit.getOAuthHelper().getRefreshToken() != null) {
```

Comment:
I should have documented why OAuthHelper also needs to be polled, but I cannot recall the reason anymore :(